server/src/main/java/org/opensearch/tasks/TaskManager.java

```diff
@@ -662,7 +663,7 @@ public CancellableTask getTask() {
 
         synchronized void registerChildNode(DiscoveryNode node) {
             if (banChildren) {
-                throw new TaskCancelledException("The parent task was cancelled, shouldn't start any child tasks");
+                throw new TaskCancelledException("The parent task was cancelled, shouldn't start any child tasks, reason:" + banReason);
```

Comment:
Nitpicking, how about we omit `reason:` and just append `banReason` when present, so `tasks, " + banReason`.